topology/graph/elasticsearch.go

```diff
@@ -250,17 +255,65 @@ func (b *ElasticSearchBackend) NodeDeleted(n *Node) bool {
 	return true
 }
 
+// GetNode get a node within a time slice
+func (b *ElasticSearchBackend) GetNode(i Identifier, t GraphContext) []*Node {
+	index := ""
+	if t.TimeSlice == nil {
+		index = b.client.GetIndexAlias()
+	}
+	nodes := b.searchNodes(&TimedSearchQuery{
+		SearchQuery: filters.SearchQuery{
+			Filter: filters.NewTermStringFilter("ID", string(i)),
+			Sort:   true,
+			SortBy: "Revision",
+		},
+		TimeFilter: getTimeFilter(t.TimeSlice),
+	}, index)
+
+	if len(nodes) > 1 && t.TimePoint {
+		return []*Node{nodes[len(nodes)-1]}
+	}
+
+	return nodes
+}
+
+func (b *ElasticSearchBackend) rollAndDumpTopology() error {
+	nodes := b.GetNodes(GraphContext{nil, false}, nil)
+	edges := b.GetEdges(GraphContext{nil, false}, nil)
+
+	if err := b.client.RollIndex(); err != nil {
```

Comment:
First thanks for addressing this feature. 

I'm not sure about the approach of creating a new index, then recreating nodes/edges because there is no guaranty to not have "partial update" in the client chan for nodes/edges that won't be present yet. I tend to think that it would be better to create "archived indexes" moving "archived records" so that we don't have to block anything neither having to take care about consistency. It could even be a new skydive tool or command line option ?

Reply:
The entire point with this PR is to allow us to delete indexes, so that we can limit the amount of storage taken by elastic search over time. I am not an expert in elastic, so don't know what "archived indexes" will provide, in any even we need to delete some indexes over time. Since there is strong reliance on history in nodes, I do not see how  "archived records" helps. The idea was that each new index is a "snapshot" standalone point , one that doesn't require historical info at all. If you can elaborate on a different approach to solve it will be good.

Reply:
my point was to move from the current index records marked as "archived" with the "archivedAt" field set to another index. The indexes where we put the "archived record" can be deleted according to a retain policy. 

My concern with the current approach is to have "partial update" in the ES client chan modifying record not already re-inserted. Maybe I missed something here, but it seems to me that the patch creates a new index and then copy active nodes from the previous index to the new one ? if so one can updates between index creation and nodes insertions ?